packages/react-heartwood-components/src/components/TruncatedList/TruncatedList-story.js

```diff
@@ -0,0 +1,230 @@
+
+		if (props.canSelect === 'one') {
+			selectedIds = sampleSize(selectedIds, 1)
+		} else {
+			selectedIds = sampleSize(selectedIds, Math.floor(selectedIds.length / 2))
+		}
+
+		const unselectedIds = props.recordSelectionListItems
+			.map(loc => loc.id)
+			.filter(locationId => selectedIds.indexOf(locationId) === -1)
+
+		const unselectableIds = props.canSelect
+			? sampleSize(unselectedIds, unselectedIds.length / 2)
+			: []
+
+		this.state = {
+			selectedIds,
+			loadedRecordSelectionListItems: cloneDeep(props.recordSelectionListItems),
+			unselectableIds
+		}
+	}
+
+	render() {
+		const {
+			header,
+			canSelect,
+			canRemove,
+			noItemsText,
+			truncatedActionItemType
+		} = this.props
+		const {
+			selectedIds,
+			loadedRecordSelectionListItems,
+			unselectableIds
+		} = this.state
+
+		return (
+			<TruncatedList
+				header={header}
+				noItemsText={noItemsText}
+				selectedIds={selectedIds}
+				unselectableIds={unselectableIds}
+				recordSelectionListItems={loadedRecordSelectionListItems.slice(0, 5)}
+				isTruncated={loadedRecordSelectionListItems.length > 5}
+				canSelect={canSelect}
+				canRemove={canRemove}
+				onSelect={id => {
+					// Typically you'd want "many" or "one", so you'd only need one side
+					// of this conditional.
+					if (canSelect === 'many') {
+						if (selectedIds.indexOf(id) >= 0) {
+							this.setState({
+								selectedIds: selectedIds.filter(selectedId => selectedId !== id)
+							})
+						} else {
+							this.setState({ selectedIds: [...selectedIds, id] })
+						}
+					} else if (canSelect === 'one') {
+						this.setState({ selectedIds: [id] })
+					}
+				}}
+				onRemove={id => {
+					// The component maintains state of the records it has loaded, but
+					// it's up to you to remove them from your list in your local state.
+					// Also, if for some reason you want to have selection alongside deletion,
+					// you should be sure to clear out the selection at the same time.
+					this.setState({
+						selectedIds: selectedIds.filter(selectedId => selectedId !== id),
+						loadedRecordSelectionListItems: loadedRecordSelectionListItems.filter(
+							locationItem => locationItem.id !== id
+						)
+					})
+				}}
+				truncatedActionText={`See all${
+					truncatedActionItemType
+						? ` ${
+								loadedRecordSelectionListItems.length
+						  } ${truncatedActionItemType}`
+						: ''
+				}`}
+				onClickTruncatedAction={() =>
+					console.log('Clicked truncated list action!')
+				}
+			/>
+		)
+	}
+}
+
+stories
+	.add('Truncated List - Single', () => {
+		return (
+			<div style={{ width: '320px', padding: '8px' }}>
+				<Card isSmall>
+					<CardHeader
+						title="Locations"
+						actions={object('actions', [
+							{
+								text: 'Add a location',
+								kind: 'simple'
+							}
+						])}
+					/>
+					<CardBody>
+						<TruncatedListExample
+							header={text('header', 'Locations')}
```

Comment:
Can we take `header` out of this example? I wouldn't recommend using it when it's redundant with the card header title. Just wanna make sure this doesn't create confusion.

Reply:
👍